session/storage_memory.go

```diff
@@ -24,35 +24,57 @@ import (
 // NewStorageMemory initiates new session storage
 func NewStorageMemory() *StorageMemory {
 	return &StorageMemory{
-		sessionMap: make(map[ID]Session),
-		lock:       sync.Mutex{},
+		sessions:  make([]Session, 0),
+		idToIndex: make(map[ID]int),
+		lock:      sync.Mutex{},
 	}
 }
 
-// StorageMemory maintains a map of session id -> session
+// StorageMemory maintains all currents sessions in memory
 type StorageMemory struct {
-	sessionMap map[ID]Session
-	lock       sync.Mutex
+	sessions  []Session
```

Comment:
I have a feeling that storing both array and map is overkill here. 
How many session we can expect to be stored here, 100 or even 1000? It will be very fast to walk through all of them to find the required one. 

Now we have to fill both array and map, that looks a bit complicated and will be broken one day: 
```golang
sessionIndex := len(storage.sessions)
storage.sessions = append(storage.sessions, sessionInstance)

storage.idToIndex[sessionInstance.ID] = sessionIndex
```

Reply:
`sessions` - It's just a cache for `FindAll()`
`idToIndex` - is cache optimisation specially for functions `Find()` & `Remove()`

I care just about abstraction in interface & anybody can come and write better implementation for this interface.
Tests are to supervise contract of abstraction